tests/test_jira.py

```diff
@@ -0,0 +1,437 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alberto Martín <alberto.martin@bitergia.com>
+#     Quan Zhou <quan@bitergia.com>
+#
+
+import argparse
+import json
+import unittest
+import shutil
+import sys
+import tempfile
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import BackendError, CacheError, ParseError
+from perceval.backends.jira import Jira, JiraClient, JiraCommand
+from perceval.utils import str_to_datetime
+
+
+JIRA_SERVER_URL = 'http://example.com'
+JIRA_SEARCH_URL = JIRA_SERVER_URL + '/rest/api/2/search'
+
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+
+class TestJiraBackend(unittest.TestCase):
+    """Jira backend tests"""
+
```

Comment:
The only thing I miss here is a test where you check the initialization values.

For instance: https://github.com/grimoirelab/perceval/blob/master/tests/test_bugzillarest.py#L122